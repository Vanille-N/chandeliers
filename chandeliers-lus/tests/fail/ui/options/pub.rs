#![allow(unused_imports)]

mod vis {
    chandeliers_lus::decl! {
        node private() returns ();
        let tel;

        const PRIVATE: int = 0;

        #[export]
        node public() returns ();
        let tel;

        #[export]
        const PUBLIC : int = 0;
    }
}

use vis::private;
use vis::PRIVATE;

use vis::public;
use vis::PUBLIC;

fn main() {}
