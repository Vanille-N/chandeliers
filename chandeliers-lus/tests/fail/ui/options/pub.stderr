error[E0603]: struct `private` is private
  --> tests/fail/ui/options/pub.rs:21:10
   |
21 | use vis::private;
   |          ^^^^^^^ private struct
   |
note: the struct `private` is defined here
  --> tests/fail/ui/options/pub.rs:6:14
   |
6  |         node private() returns ();
   |              ^^^^^^^

error[E0603]: constant `PRIVATE` is private
  --> tests/fail/ui/options/pub.rs:22:10
   |
22 | use vis::PRIVATE;
   |          ^^^^^^^ private constant
   |
note: the constant `PRIVATE` is defined here
  --> tests/fail/ui/options/pub.rs:10:15
   |
10 |         const PRIVATE: int = 0;
   |               ^^^^^^^
